packages/@ember/-internals/runtime/lib/mixins/array.d.ts

```diff
@@ -107,4 +107,4 @@ export function uniqBy<T>(
   array: T[] | EmberArray<T>,
   keyOrFunc: string | ((item: T) => unknown)
 ): T[] | EmberArray<T>;
-export function isArray(obj: unknown): boolean;
+export function isArray(obj: unknown): obj is ArrayLike<unknown> | EmberArray<unknown>;
```

Comment:
This is pretty much what is actually checked. However, the actual code is kind of bonkers and we should probably stop using it!